docs/docs/examples/secure-api/pylon.mdx

```diff
@@ -0,0 +1,280 @@
+    this.id = id;
+    this.name = name;
+    this.#messages = messages;
+  }
+
+  @requireAuth({ roles: ["read:messages"] })
+  messages() {
+    return this.#messages;
+  }
+
+  static users: User[] = [];
+
+  @requireAuth()
+  static async me() {
+    const ctx = getContext();
+    const id = ctx.get("auth")!.sub;
+
+    return User.users.find((user) => user.id === id);
+  }
+
+  @requireAuth()
+  static async create() {
+    const ctx = getContext();
+
+    const auth = ctx.get("auth")!;
+
+    // Check if the user already exists
+
+    if (User.users.find((user) => user.id === auth.sub)) {
+      throw new ServiceError("User already exists", {
+        statusCode: 400,
+        code: "USER_ALREADY_EXISTS",
+      });
+    }
+
+    const user = new User(auth.sub, auth.username || "unknown", []);
+
+    User.users.push(user);
+
+    return user;
+  }
+}
+
+export default defineService({
+  Query: {
+    me: User.me,
+    info: () => "Public Data",
+  },
+  Mutation: {
+    createUser: User.create,
+  },
+});
+
+export const configureApp: PylonAPI["configureApp"] = (app) => {
+  // Initialize the authentication middleware
+  app.use("*", auth.initialize());
+
+  app.get("/api/info", (c) => {
+    return new Response("Public Data");
+  });
+
+  app.get("/api/me", auth.require(), (c) => {
+    const id = c.get("auth")!.sub;
+
+    const user = User.users.find((user) => user.id === id);
+
+    return c.json(user);
+  });
+
+  // A role check for `read:messages` is not required here, as the `user.messages` method already checks for it.
+  app.get("/api/me/mesages", auth.require({ roles: [] }), (c) => {
+    const id = c.get("auth")!.sub;
+
+    const user = User.users.find((user) => user.id === id);
+
+    if (!user) {
+      throw new HTTPException(404, { message: "User not found" });
+    }
+
+    // This will throw an error if the user does not have the `read:messages` role
+    return c.json(user.messages());
+  });
+};
+```
+
+### Call the API
+
+To call the API you need an access token, which is then verified by ZITADEL.
+Please follow [this guide here](/docs/guides/integrate/token-introspection/private-key-jwt#get-an-access-token), ignoring the first step as we already have the `.json`-key-file from the serviceaccount.
+
+:::info
+You can also create a PAT for the serviceuser and use it to test the API. For this, follow [this guide](/docs/guides/integrate/service-users/personal-access-token#create-a-service-user-with-a-pat).
+:::
+
+Optionally set the token as an environment variable:
+
+```
+export TOKEN='MtjHodGy4zxKylDOhg6kW90WeEQs2q...'
+```
+
+With the access token, you can then do the following calls:
+
+1. GraphQL:
+
+```
+curl -H "Authorization: Bearer $TOKEN" -X GET http://localhost:3000/graphql?query={info}
+curl -H "Authorization: Bearer $TOKEN" -X GET http://localhost:3000/graphql?query={me{id name}}
+curl -H "Authorization: Bearer $TOKEN" -X GET http://localhost:3000/graphql?query={me{id name messages}}
```

Comment:
All these curl commands results in errors. It seems that curl treats the braces `{}` specially. I tried single-quoting without much luck.

```bash
curl -H "Authorization: Bearer $TOKEN" -X GET http://localhost:3000/graphql?query={info}
{"errors":[{"message":"Syntax Error: Unexpected Name \"info\".","locations":[{"line":1,"column":1}]}]
```

```bash
curl -H "Authorization: Bearer $TOKEN" -X GET http://localhost:3000/graphql?query={me{id name}}
curl: (3) nested brace in URL position 40:
http://localhost:3000/graphql?query={me{id
                                       ^
```

```bash
curl -H "Authorization: Bearer $TOKEN" -X GET http://localhost:3000/graphql?query={me{id name messages}}
curl: (3) nested brace in URL position 40:
http://localhost:3000/graphql?query={me{id
```

My curl:

```bash
curl --version
curl 7.88.1 (x86_64-pc-linux-gnu) libcurl/7.88.1 OpenSSL/3.0.11 zlib/1.2.13 brotli/1.0.9 zstd/1.5.4 libidn2/2.3.3 libpsl/0.21.2 (+libidn2/2.3.3) libssh2/1.10.0 nghttp2/1.52.0 librtmp/2.3 OpenLDAP/2.5.13
Release-Date: 2023-02-20, security patched: 7.88.1-10+deb12u5
Protocols: dict file ftp ftps gopher gophers http https imap imaps ldap ldaps mqtt pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli GSS-API HSTS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM NTLM_WB PSL SPNEGO SSL threadsafe TLS-SRP UnixSockets zstd
```